File: NoteApp-MVVM-Livedata-Databinding/app/src/main/java/com/kurio/mvvmtesting/viewmodel/NoteViewModel.java
package com.kurio.mvvmtesting.viewmodel;


import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.kurio.mvvmtesting.db.NoteRepo;
import com.kurio.mvvmtesting.model.Note;

import java.util.List;

public class NoteViewModel extends AndroidViewModel {
    NoteRepo noteRepo;
   public LiveData<List<Note>> allNote;

    public NoteViewModel(@NonNull Application application) {
        super(application);
        this.noteRepo = new NoteRepo(application);
        allNote = noteRepo.getAllNotes();
    }

    public void insertNote(Note note) {
        noteRepo.insertNote(note);
    }

    public void deleteNote(Note note) {
        noteRepo.deleteNote(note);
    }

    public void updateNote(Note note) {
        noteRepo.updateNote(note);
    }
}

File: LiveData (Kotlin)/app/src/main/java/com/kurio/mvvmtesting/ui/viewmodel/NoteViewModel.kt
package com.kurio.mvvmtesting.ui.viewmodel


import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData

import com.kurio.mvvmtesting.db.NoteRepo
import com.kurio.mvvmtesting.model.Note

class NoteViewModel(application: Application) : AndroidViewModel(application) {
    private val noteRepo: NoteRepo = NoteRepo(application)
    var allNote: LiveData<List<Note>>

    init {
        allNote = noteRepo.allNotes
    }

    fun insertNote(note: Note) {
        noteRepo.insertNote(note)
    }

    fun deleteNote(note: Note) {
        noteRepo.deleteNote(note)
    }

    fun updateNote(note: Note) {
        noteRepo.updateNote(note)
    }
}
